client/signup/steps/user/index.jsx

```diff
@@ -272,9 +273,15 @@ export class UserStep extends Component {
 			}
 		}
 
+		let formProps = omit( this.props, [ 'translate' ] );
+		if ( this.props.flowName === 'crowdsignal' && this.props.oauth2Client && isCrowdsignalOAuth2Client( this.props.oauth2Client ) ) {
+			formProps.displayNameInput = abtest( 'crowdsignalNameBasedSignup' ) === 'nameSignup';
+			formProps.displayUsernameInput = abtest( 'crowdsignalNameBasedSignup' ) !== 'nameSignup';
```

Comment:
Ah nice! so essentially, these are displayed conditionally! Question on those, what will be the strategy once we finish the abtest? Will we just refactor this to be the test result and then deploy?

Reply:
Yeah, as the actual step/signup-form is used in different forms we only want the test to affect the new Crowdsignal flow.

When we're done this whole part should be removed and the result should be set using the settings [here](https://github.com/Automattic/wp-calypso/blob/master/client/signup/config/steps-pure.js#L224) - which this code now overrides. This was necessary as those settings are generated at build time meaning we can't use the `abtest` library right there.